tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_cli_all_targets_lost.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+# Copyright 2023 Nordic Semiconductor
+# SPDX-License-Identifier: Apache-2.0
+
+source $(dirname "${BASH_SOURCE[0]}")/../../_mesh_test.sh
+
+# The test instance sequence must stay as it is due to addressing scheme
+# `fail-step` informs `dfu_cli_all_targets_lost` instance what targets it tests.
+conf=prj_mesh1d1_conf
+overlay=overlay_pst_conf
```

Comment:
I agree, but `test_dfu.c` is build when settings are enabled: https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bsim/bluetooth/mesh/CMakeLists.txt#L26-L39, and this is only overlay that does that. I don't think that there is a point in duplicating code into separate file that does the same but is built without persistent storage, and also requires new overlay.

Reply:
I wonder if enabled pst functionality is required at all for dfu tests but this is a general question. Not to this PR.

Reply:
I added tests for DFU Server recovery after reboot of device, these surely require pst. I guess these could be moved into persistance test folder, but this would require copying there all callbacks, composition data and so on. Maybe mesh should have `CONFIG_BT_SETTINGS` enabled, without flash? But as you said, it's not a scope of this PR